lib/rouge/lexers/logtalk.rb

```diff
@@ -0,0 +1,252 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Logtalk < RegexLexer
+      title "Logtalk"
+      desc "The Logtalk programming language (http://logtalk.org/)"
+      tag 'logtalk'
+      aliases 'logtalk'
+      filenames '*.lgt', '*.logtalk'
+      mimetypes 'text/x-logtalk'
+
+      def self.analyze_text(text)
+          return 1.0 if text.include? ':- object('
+          return 1.0 if text.include? ':- protocol('
+          return 1.0 if text.include? ':- category('
+          return 0.9 if text =~ /^:-\s[a-z]/
+      end
+
+      state :basic do
+        # Directives
+        rule /^\s*:-\s/, Keyword, :directives
+        # Whitespace
+        mixin :whitespace
+        # Comments
+        mixin :comments
+        # Numbers
+        mixin :numbers
+        # Variables
+        mixin :variables
+        # Event handlers
+        rule /(after|before)(?=[(])/, Keyword
+        # Message forwarding handler
+        rule /forward(?=[(])/, Keyword
+        # Execution-context methods
+        rule /(parameter|this|se(lf|nder))(?=[(])/, Keyword
+        # Reflection
+        rule /(current_predicate|predicate_property)(?=[(])/, Keyword
+        # DCGs and term expansion
+        rule /(expand_(goal|term)|(goal|term)_expansion|phrase)(?=[(])/, Keyword
+        # Entity
+        rule /(abolish|c(reate|urrent))_(object|protocol|category)(?=[(])/, Keyword
+        rule /(object|protocol|category)_property(?=[(])/, Keyword
+        # Entity relations
+        rule /co(mplements_object|nforms_to_protocol)(?=[(])/, Keyword
+        rule /extends_(object|protocol|category)(?=[(])/, Keyword
+        rule /imp(lements_protocol|orts_category)(?=[(])/, Keyword
```

Comment:
Hey, so I'm pretty sure the regex implementation is going to make this optimization for us, so I'd rather the full words were just spelled out in these cases.